core/resolve/src/mill/resolve/ParseArgs.scala

```diff
@@ -92,9 +92,49 @@ private[mill] object ParseArgs {
       : Result[(Option[Segments], Option[Segments])] =
     parse(selectorString, selector(using _)) match {
       case f: Parsed.Failure => Result.Failure(s"Parsing exception ${f.msg}")
-      case Parsed.Success(selector, _) => Result.Success(selector)
+      case Parsed.Success(selector, _) =>
+        // Post-process the segments to handle .super suffix
+        val processedSelector = processSuperSuffix(selector)
+        Result.Success(processedSelector)
     }
 
+  private def processSuperSuffix(selector: (
```

Comment:
We don't need to process the first element in the tuple IIRC, since that's the name of the `ExternalModule` (if any). Processing the second `Option[Segments]` is sufficient

Reply:
This whole method is very strange; you are taking the `Option[Segments]`, taking the separate `.super` segment and packing it into the preceding segment as part of a string, and then un-packing the string later in `instantiateSuperTask` to extract the values you care about. Can we skip this whole process and have `instantiateSuperTask` just handle the original `Option[Segments]` directly?